### ISSUE TITLE
Twitter crawling method

### ISSUE THREAD

**Issue by panhartstuff** [KurtBestor/Hitomi-Downloader#1546]
If you don't mind can you explain your method of crawling Twitter? I'm trying to get a complete crawl using several methods, but the result is just not as complete as your tool.

edit:
I read through this issue: https://github.com/KurtBestor/Hitomi-Downloader-issues/issues/1155 and tried out the methods, but I still can't get certain tweets that your tool managed to get.

For example, this particular tweet: https://twitter.com/udon0531/status/785302504653197312
I couldn't access it even through the advanced search (nor the media tab): https://twitter.com/search?q=(from%3Audon0531)%20max_id%3A867420736897531905%20exclude%3Aretweets%20filter%3Amedia%20-filter%3Aperiscope&src=typed_query

I tried a variety of other searches and I couldn't get it either. How did your tool retrieve that?

**Comment by KurtBestor** [comment]
Twitter returns different results depending on the user agent. If the results are not enough, the program tries to change the user agent several times to get more results.

**Comment by panhartstuff** [comment]
Can you please clarify the list of user agents it cycles through?

**Comment by KurtBestor** [comment]
It's randomly picked from [fake-useragent](https://github.com/hellysmile/fake-useragent).

**Comment by anonieee** [comment]
Thank you for the explanation! I will try this out.
But if you don't mind can you explain a little bit more?
How do you determine if the results are not enough? Also do you use `ua.random`?

**Comment by panhartstuff** [comment]
Thank you for your help! I will try this out.
But if you don't mind, can you explain a little more about how you determine if the results are not enough? Also do you use `ua.random`?

**Comment by KurtBestor** [comment]
If the number of collected tweets is not changed several times, it'll assume that's enough.

I don't use `ua.random` directly, but it would be alright.

**Comment by panhartstuff** [comment]
Thank you, I managed to get the missing tweets via user agent switching.

I have another question, it seems that sometimes you don't get all the way to the end via a single search. So when I reach the end of the search timeline, I get the id of the last tweet and use it in "max_id:xxxx" for a second search pass. I was able to get more tweets this way. But I'm not sure if this is the proper way.

For example, with searching: "https://twitter.com/search?q=from%3Audon0531%20exclude%3Aretweets%20filter%3Amedia%20-filter%3Aperiscope&src=typed_query&f=live"
I was missing a few of their older tweets. So I needed to get the last id of the tweet from that search and do a max_id search:
"https://twitter.com/search?q=from%3Audon0531%20max_id%3A260351044213698560%20exclude%3Aretweets%20filter%3Amedia%20-filter%3Aperiscope&src=typed_query&f=live"

Is this what you are doing?  Because I was still missing some older tweets when compared to your app's results.
